scripts/sign.py

```diff
@@ -4,55 +4,99 @@
 #
 # SPDX-License-Identifier: BSD-2-Clause
 
-
 def uuid_parse(s):
     from uuid import UUID
     return UUID(s)
 
-
 def int_parse(str):
     return int(str, 0)
 
 
 def get_args():
+    import sys
+    print(sys.argv)
     from argparse import ArgumentParser
-
-    parser = ArgumentParser()
+    command_choices = ['sign', 'generate-digest', 'stitch-ta']
+
+    parser = ArgumentParser(description='Sign a Tusted Application for OP-TEE.',usage='\n   ' +
+        '%(prog)s command [ command_opts ] command_args\n\n'
+
+    '   command:\n' +
+    '     sign             Generate signed loadable TA image file.\n' +
+    '                      Takes arguments --uuid, --in, --out and --key.\n' +
+    '     generate-digest  Generate loadable TA binary image digest for offline\n' +
+    '                      signing. Takes arguments  --uuid, --in and --out.\n' +
+    '     stitch-ta        Generate loadable signed TA binary image file from\n' +
+    '                      TA raw image and its signature. Takes arguments\n' +
+    '                      --uuid, --in, --out, and --sig.\n\n' +
+
+    'command_opts\n' +
+    '    --version NUM    NUM shall be 0. Defaults to 0.\n\n' +
+
+    'command_arg\n' +
+    '    --uuid UUID      String UUID of the TA\n' +
+    '    --in INFILE      Input raw unsigned TA image file path\n' +
+    '    --out OUTFILE    Generated output file path\n' +
+    '    --key PRIVKEY    Private key in PEM format\n' +
+    '    --sig SIGNATURE  TA image signature binary file\n' +
+    '    --dig DIGEST     TA image digest binary file\n',
+               epilog='If no command is given, the script will default to "sign".')
+
+    # when choices= is used, the command can no longer be left out and the default used
```

Comment:
Minor: please start all sentences with an uppercase letter in the inline comments and commit comments.
I mean in most optee_os file tree.